coredns/tests/test_check.py

```diff
@@ -0,0 +1,170 @@
+# (C) Datadog, Inc. 2010-2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+import os
+import mock
+import pytest
+import requests
+import subprocess
+
+# project
+from datadog_checks.coredns import CoreDNSCheck
+from datadog_checks.utils.common import get_docker_hostname
+
+instance = {
+    'prometheus_endpoint': 'http://localhost:9153/metrics',
+}
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+COREDNS_HOST = get_docker_hostname()
+COREDNS_PORT = '9153'
+
+DOCKER_COMPOSE_ARGS = [
+    "docker-compose",
+    "-f", os.path.join(HERE, 'docker', 'docker-compose.yaml')
+]
+
+DIG_ARGS = [
+    "dig",
+    "google.com",
+    "@127.0.0.1",
+    "-p",
+    "54"
+]
+
+class MockResponse:
+    """
+    MockResponse is used to simulate the object requests.Response commonly returned by requests.get
+    """
+
+    def __init__(self, content, content_type):
+        self.content = content
+        self.headers = {'Content-Type': content_type}
+
+    def iter_lines(self, **_):
+        for elt in self.content.split("\n"):
+            yield elt
+
+    def raise_for_status(self):
+        pass
+
+    def close(self):
+        pass
+
+@pytest.fixture
+def coredns():
+    env = os.environ
```

Comment:
instead of calling subprocess command to start your container, you can leverage utilities in this [file](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_dev/datadog_checks/dev/docker.py#L46). You can find good examples [here](https://github.com/DataDog/integrations-core/blob/master/zk/tests/conftest.py#L111) or [there](https://github.com/DataDog/integrations-core/blob/master/sqlserver/tests/conftest.py#L81). Feel free to take a look at what the `datadog_checks_dev` package contains since it holds a lot of useful methods :)